### PR TITLE
[uss_qualifier] Directly query Service Providers in RID nominal behavior test

### PR DESCRIPTION
**Pull request by BenjaminPelletier**
Currently, the RID nominal behavior test is completely independent of which RID standard is used because it only tests whether injections cause observations.  The down side of this approach is that the responsible party (between the Service Provider and Display Provider) cannot easily be assigned.  This PR adds an optional feature to that scenario that will cause uss_qualifier to directly query the DSS and Service Provider when DSS information is provided.  The responses to these queries are checked first before the end-to-end observation is performed.

### CODE REVIEW

File: monitoring/uss_qualifier/scenarios/astm/netrid/nominal_behavior.md [hunk 1]
```diff
@@ -41,9 +45,39 @@ Per **[interuss.automated_testing.rid.injection.UpsertTestResult](../../../requi
...
+
+**[astm.f3411.v19.NET0710](../../../requirements/astm/f3411/v19.md)** requires a Service Provider to implement the GET flight details endpoint.  This check will fail if uss_qualifier cannot query that endpoint (specified in the ISA present in the DSS) successfully.
+
+#### Lingering flight check
+
+**[astm.f3411.v19.NET0260](../../../requirements/astm/f3411/v19.md)** requires a SP to provide flights up to *NetMaxNearRealTimeDataPeriod* in the past, but an SP should preserve privacy and ensure relevancy by not sharing flights that are further in the past than this window.
+
+#### Area too large check
+
+**[astm.f3411.v19.NET0430](../../../requirements/astm/f3411/v19.md)** require that a NetRID Display Provider reject a request for a very large view area with a diagonal greater than *NetMaxDisplayAreaDiagonal*.  If such a large view is requested and a 413 error code is not received, then this check will fail.
```
Comment (barroco) [inline monitoring/uss_qualifier/scenarios/astm/netrid/nominal_behavior.md:76]:
```suggestion
**[astm.f3411.v19.NET0430](../../../requirements/astm/f3411/v19.md)** requires that a NetRID Display Provider rejects a request for a very large view area with a diagonal greater than *NetMaxDisplayAreaDiagonal*.  If such a large view is requested and a 413 error code is not received, then this check will fail.
```

File: monitoring/uss_qualifier/scenarios/astm/netrid/nominal_behavior.md [hunk 2]
```diff
@@ -57,14 +91,14 @@ Per **[interuss.automated_testing.rid.observation.UniqueFlights](../../../requir
 
 The timestamps of the injected telemetry usually start in the future.  If a flight with injected telemetry only in the future is observed prior to the timestamp of the first telemetry point, this check will fail because the SP does not satisfy **[interuss.automated_testing.rid.injection.ExpectedBehavior](../../../requirements/interuss/automated_testing/rid/injection.md)**.
 
-#### Lingering flight check
-
-**[astm.f3411.v19.NET0260](../../../requirements/astm/f3411/v19.md)** requires a SP to provide flights up to *NetMaxNearRealTimeDataPeriod* in the past, but an SP should preserve privacy and ensure relevancy by not sharing flights that are further in the past than this window.
-
 #### Missing flight check
 
 **[astm.f3411.v19.NET0610](../../../requirements/astm/f3411/v19.md)** require that SPs make all UAS operations discoverable over the duration of the flight plus *NetMaxNearRealTimeDataPeriod*, so each injected flight should be observable during this time.  If one of the flights is not observed during its appropriate time period, this check will fail.
```
Comment (barroco) [inline monitoring/uss_qualifier/scenarios/astm/netrid/nominal_behavior.md:96]:
```suggestion
**[astm.f3411.v19.NET0610](../../../requirements/astm/f3411/v19.md)** requires that SPs make all UAS operations discoverable over the duration of the flight plus *NetMaxNearRealTimeDataPeriod*, so each injected flight should be observable during this time.  If one of the flights is not observed during its appropriate time period, this check will fail.
```

**Review by barroco — APPROVED** [review]
Looks good! Very minor comments inline

File: monitoring/uss_qualifier/scenarios/astm/netrid/display_data_evaluator.py [hunk 3]
```diff
@@ -354,3 +405,105 @@ def _evaluate_area_too_large_observation(
...
+    ) -> None:
+        if not dp_observation.dss_isa_query.success:
+            # Note: This step currently uses the DSS endpoint to perform a one-time query for ISAs, but this
+            # endpoint is not strictly required.  The PUT Subscription endpoint, followed immediately by the
+            # DELETE Subscription would produce the same result, but because uss_qualifier does not expose any
+            # endpoints (and therefore cannot provide a callback/base URL), calling the one-time query endpoint
+            # is currently much cleaner.  If this test is applied to a DSS that does not implement the one-time
+            # ISA query endpoint, this check can be adapted.
+            with self._test_scenario.check("ISA query") as check:
+                if not dp_observation.dss_isa_query.success:
```
[the remark this answers is not in the document]
Comment (BenjaminPelletier) [reply monitoring/uss_qualifier/scenarios/astm/netrid/display_data_evaluator.py:422]: Ah, good point -- fixed.

File: monitoring/uss_qualifier/scenarios/astm/netrid/nominal_behavior.md (hunk 1 above)
Comment (BenjaminPelletier) [reply monitoring/uss_qualifier/scenarios/astm/netrid/nominal_behavior.md:76]: Done.

File: monitoring/uss_qualifier/scenarios/astm/netrid/nominal_behavior.md (hunk 2 above)
Comment (BenjaminPelletier) [reply monitoring/uss_qualifier/scenarios/astm/netrid/nominal_behavior.md:96]: Done.

**Pull request merged**